### ISSUE TITLE
bug: Step function issue with waitForTaskToken

### ISSUE THREAD

**Issue by kelhelawy** [localstack/localstack#9883]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

After upgrading the LocalStack v3, and using the new stepfunctions provider, the following step function:
```
{
            "Parameters" : {
              "Message" : {
                "request_model.$" : "$",
                "step_id" : "decision-workflow-execute-decision-action-step",
                "task_token.$" : "$$.Task.Token",
                "execution_arn.$" : "$$.Execution.Id"
              },
              "TopicArn" : "arn:aws:sns:ap-south-1:000000000000:sf-router",
              "MessageAttributes" : {
                "routingKey" : {
                  "StringValue.$" : "$.routing_key",
                  "DataType" : "String"
                }
              }
            },
            "ResultSelector" : {
              "metadata.$" : "$.metadata",
              "trace_context_as_b3.$" : "$.trace_context_as_b3",
              "payload.$" : "States.StringToJson($.payload)",
              "resource_package_id.$" : "$.resource_package_id",
              "step_function_version.$" : "$.step_function_version",
              "request_id.$" : "$.request_id",
              "deployment_id.$" : "$.deployment_id",
              "locks.$" : "$.locks",
              "routing_key.$" : "$.routing_key",
              "current_state.$" : "$.current_state"
            },
            "End" : true,
            "Resource" : "arn:aws:states:::sns:publish.waitForTaskToken",
            "Type" : "Task"
```
results in the message only contains String of `request_model`, not the full json, While in the old provider returns the expected result:
```
{
   "body":{
      "step_id":"decision-workflow-execute-decision-action-step",
      "request_model":{
         "metadata":{
            "callbackAudience":"callbackAudience",
            "callbackUrl":"......."
         },
         "trace_context_as_b3":".....",
         "resource_package_id":"....",
         "payload":{
            
         },
         "current_state":"IN_PROGRESS",
         "step_function_version":"V000"
         ....
   }
}
```

### How are you starting LocalStack?

With a docker-compose file

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    docker run localstack/localstack

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    awslocal s3 mb s3://mybucket

### Environment

```markdown
- OS: MacOs
- LocalStack: 3.0.3
```

**Comment by kelhelawy** [comment]
Note:
I just noticed that choosing changing `request_model` to `body` make it works, but can I change it to custom name, as it works on AWS and the old provider?

```
{
   "step_id":"decision-workflow-package-acquire-lock-step",
   "task_token":"....",
   "body":{
      "metadata":{
         "callbackAudience":"callbackAudience",
         "callbackUrl":"..."
      },
      "trace_context_as_b3":"....",
      "resource_package_id":"....",
      "payload":{
         ......
      }
   },
   "execution_arn":"...."
}
```

**Comment by MEPalma** [comment]
Hello, thank you for compiling this issue. We have recently merged some changes that should address the issue reported. The next nightly release will include such fixes.
Please do feel free to reopen this issue or reach out should this not be resolved for you. Thank you!

**Comment by kelhelawy** [comment]
Thanks @MEPalma , but this doesnt mean it was released already in `3.1`, as added in the milestone correct?

**Comment by MEPalma** [comment]
Correct, it is not available in 3.1, but only from the next nightly
